### PR TITLE
Test SQL Server only on 2019-CU28-ubuntu-20.04

### PR DESCRIPTION
**Pull request by wendigo**
## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
## Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodeSqlserver.java [hunk 1]
```diff
@@ -56,7 +56,7 @@ public void extendEnvironment(Environment.Builder builder)
     @SuppressWarnings("resource")
     private DockerContainer createSqlServer()
     {
-        DockerContainer container = new DockerContainer("mcr.microsoft.com/mssql/server:2017-latest", "sqlserver")
+        DockerContainer container = new DockerContainer("mcr.microsoft.com/mssql/server:2019-CU28-ubuntu-20.04", "sqlserver")
```
Comment (Praveen2112) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodeSqlserver.java:59]: Was it fixed on 2019 ?

Comment (wendigo) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodeSqlserver.java:59]: Yes, it was fixed in 2019

Comment (Praveen2112) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodeSqlserver.java:59]: Can we update the latest to 2023 in this case ? Do we need to update the docs for the same ?

Comment (wendigo) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodeSqlserver.java:59]: Let's keep it on 2019, 2023 seems to new ;)

**Review by losipiuk — APPROVED** [review]

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodeSqlserver.java (hunk 1 above)
Comment (wendigo) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodeSqlserver.java:59]: I'll update the docs

**Review by ebyhr — APPROVED** [review]

**Comment by bind-cannon** [comment]
@wendigo The release notes for 465 suggest SQL server 2019 or higher is required for a SQL server connector. Is that true or is it that Trino is just not testing against 2017 anymore? We still use 2017 in production and this would prevent us from upgrading to the newer Trino version.

**Comment by losipiuk** [comment]
> @wendigo The release notes for 465 suggest SQL server 2019 or higher is required for a SQL server connector. Is that true or is it that Trino is just not testing against 2017 anymore? We still use 2017 in production and this would prevent us from upgrading to the newer Trino version.

It is the latter. I do not see a reason that it would stop working with older version now. But given lack of test it is hard to promise anything.

**Pull request merged**